### PR TITLE
Support gcs in parquet dataset

### PR DESCRIPTION
**Pull request by eddyxu**
Add support of GCS in `rikai.parquet.Dataset` for #20

### CODE REVIEW

**Comment by eddyxu** [comment]
Similar to #149 , only run test manually due to the lack of Google Cloud test infra.

```python
    for row in Dataset("gs://my_bucket/coco_raw/output/"):
        print(row)
        break
```

File: python/rikai/io.py [hunk 1]
```diff
@@ -116,10 +117,10 @@ def copy(source: str, dest: str) -> str:
             _gcsfs().copy(source, dest)
             return dest
 
-    # TODO: find better i/o utilis to copy between filesystems
-    with _open_output_stream(dest) as out_stream:
-        with _open_input_stream(source) as in_stream:
-            shutil.copyfileobj(in_stream, out_stream)
+    with _open_output_stream(dest) as out_stream, open_input_stream(
```
Comment (eddyxu) [inline python/rikai/io.py:120]: was formatted by `black`

**Comment by eddyxu** [comment]
Filed #154 for future cloud integration test.

**Review by changhiskhan — APPROVED** [review]
please make sure aws integration still works. otherwise looks good.

**Comment by eddyxu** [comment]
```
>>> from rikai.parquet import Dataset
>>> d = Dataset("s3://rikai-test/output")
2021-03-27 14:10:09,200 INFO Rikai (dataset.py:111): Loading parquet files: ['s3://rikai-test/output/part-00000-3ed9d7e4-1608-495d-b018-27b4340d9388-c000.snappy.parquet', 's3://rikai-test/output/part-00001-3ed9d7e4-1608-495d-b018-27b4340d9388-c000.snappy.parquet', 's3://rikai-test/output/part-00002-3ed9d7e4-1608-495d-b018-27b4340d9388-c000.snappy.parquet', 's3://rikai-test/output/part-00003-3ed9d7e4-1608-495d-b018-27b4340d9388-c000.snappy.parquet', 's3://rikai-test/output/part-00004-3ed9d7e4-1608-495d-b018-27b4340d9388-c000.snappy.parquet', 's3://rikai-test/output/part-00005-3ed9d7e4-1608-495d-b018-27b4340d9388-c000.snappy.parquet']
>>> next(iter(d))
{'image_id': 391895, 'annotations': [{'category_id': 4, 'category_text': 'motorcycle', 'area': 12190.4453125, 'bbox': Box2d(xmin=359.17, ymin=146.17, xmax=471.62, ymax=359.74)}, {'category_id': 1, 'category_text': 'person', 'area': 14107.271484375, 'bbox': Box2d(xmin=339.88, ymin=22.16, xmax=493.76, ymax=322.89000000000004)}, {'category_id': 1, 'category_text': 'person', 'area': 708.2605590820312, 'bbox': Box2d(xmin=471.64, ymin=172.82, xmax=507.56, ymax=220.92)}, {'category_id': 2, 'category_text': 'bicycle', 'area': 626.9852294921875, 'bbox': Box2d(xmin=486.01, ymin=183.31, xmax=516.64, ymax=218.29)}], 'image': Image(uri=/home/lei/datasets/coco/train2017/000000391895.jpg), 'split': 'train'}
```

**Pull request merged**